daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -102,7 +85,7 @@ object VersionedTransaction extends value.CidContainer2[VersionedTransaction] {
   * Therefore, it is '''forbidden''' to create ill-formed instances, i.e., instances with `!isWellFormed.isEmpty`.
   */
 final case class GenTransaction[Nid, +Cid, +Val](
-    nodes: HashMap[Nid, Node.GenNode[Nid, Cid, Val]],
+    nodes: Map[Nid, Node.GenNode[Nid, Cid, Val]],
```

Comment:
What made you change this?

Reply:
that allows me to write: 

```scala
 override def nodes: Map[Nid, Node.GenNode.WithTxValue[Nid, Cid]] =
    versionedNodes.mapValues(_.node)

  def transaction: GenTransaction[Nid, Cid, Transaction.Value[Cid]] =
    GenTransaction(nodes, roots)
```

Because, the method `mapValues` "wraps the original map without copying any elements" it does not return a `HashMap`. 
It is in general a bit dangerous because the values of the resulting map, are recomputed each time the map is looked up. In this case I think it is fine because the function `_.node` is very cheap. 
I will add a comment to the method `nodes`.

With the change the methods `transaction` are constant time. I could not achieve this if I did not change the type of `GenTransaction`'s  `nodes` field.

Reply:
I understand. That makes sense to me. Thanks for the explanation.